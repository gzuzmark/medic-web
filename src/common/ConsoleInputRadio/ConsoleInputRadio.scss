@import "../../assets/styles/01-settings/settings";

.ConsoleInputRadio {
  cursor: pointer;
  display: flex;
  font-size: 22px;
  margin-bottom: 12px;
  padding-left: 25px;
  position: relative;
  &--input {
    cursor: pointer;
    opacity: 0;
    position: absolute;
    &:checked {
      &~ .ConsoleInputRadio--radio {
        background-color: white;
        &:after {
          background: #4a4a4a;
        }
      }
    }
  }
  &--radio {
    background-color: white;
    border: 1px solid #ccd6dc;
    border-radius: 50%;
    height: 22px;
    left: 0;
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    width: 22px;
    &:after {
      background: white;
      border-radius: 50%;
      content: "";
      height: 10px;
      left: 50%;
      position: absolute;
      top: 50%;
      transform: translate(-50%, -50%);
      transition: background ease-in 0.2s;
      width: 10px;
    }
  }
  user-select: none;
}
