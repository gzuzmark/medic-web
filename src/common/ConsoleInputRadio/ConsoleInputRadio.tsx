import * as React from 'react';
import { Text } from '../ConsoleText';
import './ConsoleInputRadio.scss';

interface IPropsConsoleInputRadio {
    title: string;
    name: string;
    checked: boolean;
    value: string;
    onChange: (event?: any) => void
}
const ConsoleInputRadio: React.StatelessComponent<IPropsConsoleInputRadio> = (props) => {
    return (
        <label className="ConsoleInputRadio">
            <Text style={{fontSize: 16}}>{props.title}</Text>
            <input className="ConsoleInputRadio--input"
                   type="radio"
                   name={props.name}
                   value={props.value}
                   checked={props.checked}
                   onChange={props.onChange}/>
            <span className="ConsoleInputRadio--radio">&nbsp;</span>
        </label>
    );
};

export default ConsoleInputRadio;