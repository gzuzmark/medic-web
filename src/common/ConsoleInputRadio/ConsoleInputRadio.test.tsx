import { shallow } from 'enzyme';
import * as React from 'react';
import ConsoleInputRadio from "./ConsoleInputRadio";


describe('ConsoleInputRadio Test',() => {
    let props: any;
    let mountedConsoleInputRadio: any;
    const getComponent = () => {
        if (!mountedConsoleInputRadio) {
            mountedConsoleInputRadio = shallow(
                <ConsoleInputRadio {...props} />
            );
        }
        return mountedConsoleInputRadio;
    };

    beforeEach(() => {
        props = {
            checked: false,
            name: 'demo',
            onChange: () => void(0),
            title: 'Demo',
            value: "value",
        };
        mountedConsoleInputRadio = undefined;
    });



    it("render: render ConsoleInputRadio", () => {
        const component = getComponent();
        expect(component).toMatchSnapshot();
    });
});