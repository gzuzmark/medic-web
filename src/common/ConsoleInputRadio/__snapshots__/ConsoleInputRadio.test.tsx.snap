// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsoleInputRadio Test render: render ConsoleInputRadio 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConsoleInputRadio
    attrs={
      Object {
        "checked": false,
        "name": "demo",
        "onChange": [Function],
        "value": "value",
      }
    }
    title="Demo"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <styled.span
          style={
            Object {
              "display": "block",
              "marginLeft": 5,
            }
          }
          weight="300"
        >
          Demo
        </styled.span>,
        <input
          checked={false}
          className="ConsoleInputRadio--input"
          name="demo"
          onChange={[Function]}
          type="radio"
          value="value"
        />,
        <span
          className="ConsoleInputRadio--radio"
        >
           
        </span>,
      ],
      "className": "ConsoleInputRadio",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Demo",
          "style": Object {
            "display": "block",
            "marginLeft": 5,
          },
          "weight": "300",
        },
        "ref": null,
        "rendered": "Demo",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": false,
          "className": "ConsoleInputRadio--input",
          "name": "demo",
          "onChange": [Function],
          "type": "radio",
          "value": "value",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " ",
          "className": "ConsoleInputRadio--radio",
        },
        "ref": null,
        "rendered": " ",
        "type": "span",
      },
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.span
            style={
              Object {
                "display": "block",
                "marginLeft": 5,
              }
            }
            weight="300"
          >
            Demo
          </styled.span>,
          <input
            checked={false}
            className="ConsoleInputRadio--input"
            name="demo"
            onChange={[Function]}
            type="radio"
            value="value"
          />,
          <span
            className="ConsoleInputRadio--radio"
          >
             
          </span>,
        ],
        "className": "ConsoleInputRadio",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Demo",
            "style": Object {
              "display": "block",
              "marginLeft": 5,
            },
            "weight": "300",
          },
          "ref": null,
          "rendered": "Demo",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": false,
            "className": "ConsoleInputRadio--input",
            "name": "demo",
            "onChange": [Function],
            "type": "radio",
            "value": "value",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " ",
            "className": "ConsoleInputRadio--radio",
          },
          "ref": null,
          "rendered": " ",
          "type": "span",
        },
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
